blurring_as_a_service/inference_pipeline/submit_inference_pipeline.py

```diff
@@ -13,61 +14,64 @@
-        )
-
-        input_root_folder = Input(
-            type=AssetTypes.URI_FOLDER,
-            path=blob_container_path,
-            description="Data to be blurred",
-        )
-
-        # Get the txt file that contains all paths of the files to run inference on
-        files_to_blur_path = customer['inputs']['files_to_blur'].format(
-            subscription=subscription_id,
-            resourcegroup=resource_group,
-            workspace=workspace_name,
-            datastore_name=f"{customer_name}_input_structured"
-        )
-
-        files_to_blur_txt = Input(
-            type=AssetTypes.URI_FILE,
-            path=files_to_blur_path,
-            description="Data to be blurred",
-        )
-
-        model_parameters = customer['model_parameters']
-        model_parameters_json = json.dumps(model_parameters)  # TODO it seems I can not pass a string to @command_component function
-
-        detect_and_blur_sensitive_data_step = detect_and_blur_sensitive_data(
-            mounted_root_folder=input_root_folder,
-            relative_paths_files_to_blur=files_to_blur_txt,
-            customer_name=customer_name,
-            model_parameters_json=model_parameters_json
-        )
-
-        azureml_outputs_formatted = customer['outputs']['results_path'].format(
-            subscription=subscription_id,
-            resourcegroup=resource_group,
-            workspace=workspace_name,
-            datastore_name=f"{customer_name}_output"
-        )
-
-        detect_and_blur_sensitive_data_step.outputs.results_path = Output(
-            type="uri_folder", mode="rw_mount", path=azureml_outputs_formatted
-        )
-        detect_and_blur_sensitive_data_step.outputs.yolo_yaml_path = Output(
-            type="uri_folder", mode="rw_mount", path=blob_container_path
-        )
+    # Call .result() to get the actual values
+    workspace_name_actual = workspace_name.result()
+    subscription_id_actual = subscription_id.result()
+    resource_group_actual = resource_group.result()
+
+    customer_name = inference_settings['customer_name']
+
+    # Format the root path of the Blob Storage Container in Azure using placeholders
+    blob_container_path = inference_settings['container_root'].format(
+        subscription=subscription_id_actual,
+        resourcegroup=resource_group_actual,
+        workspace=workspace_name_actual,
+        datastore_name=f"{customer_name}_input_structured"
+    )
+
+    input_root_folder = Input(
+        type=AssetTypes.URI_FOLDER,
+        path=blob_container_path,
+        description="Data to be blurred",
+    )
+
+    # Get the txt file that contains all paths of the files to run inference on
+    files_to_blur_path = inference_settings['inputs']['files_to_blur'].format(
+        subscription=subscription_id_actual,
+        resourcegroup=resource_group_actual,
+        workspace=workspace_name_actual,
+        datastore_name=f"{customer_name}_input_structured"
+    )
```

Comment:
why the formatting?
I thought we just pass a url here.
If you wanna keep it like this I recommend we add constraints in the settings which are checking the config file.
Because now this formatting only happens in the inference pipeline

Reply:
We want to use "data assets" later in all pipelines. We created a ticket for that: https://gemeente-amsterdam.atlassian.net/browse/BCV-207